SpellGUIV2/MainWindow.xaml.cs

```diff
@@ -2926,47 +2922,22 @@ private void ResizeChildElements(double xChange, double yChange, FrameworkElemen
         }
         #endregion
 
-        public void changeConfigValue(string key, string value)
-        {
-            if (!File.Exists("config.xml"))
-                return;
-
-            var xml = new XmlDocument();
-            xml.Load("config.xml");
-            var node = xml.SelectSingleNode("MySQL/" + key);
-
-            if (node == null)
-                return;
-
-            node.InnerText = value;
-            xml.Save("config.xml");
-
-            if (GetConfig() != null)
-                config.ReadConfigFile();
-        }
-
-        public string getConfigValue(string key)
-        {
-            var xml = new XmlDocument();
-            if (!File.Exists("config.xml"))
-                return "";
-
-            xml.Load("config.xml");
-            var node = xml.SelectSingleNode("MySQL/" + key);
-            return node==null ? "" : node.InnerText;
-        }
-
         private void MultilingualSwitch_SelectionChanged(object sender, SelectionChangedEventArgs e)
         {
             string language = e.AddedItems[0].ToString();
-            string path = string.Format("pack://{0}:,,,/Languages/{1}.xaml", language == "enUS" ? "application" : "SiteOfOrigin", language);
+            string path = string.Format("pack://{0}:,,,/Languages/{1}.xaml", "SiteOfOrigin", language);
             Application.Current.Resources.MergedDictionaries[0].Source = new Uri(path);
-            changeConfigValue("language", language);
+            config.UpdateConfigValue("language", language);
         }
 
         private void MultilingualSwitch_Initialized(object sender, EventArgs e)
         {
-            string configLanguage = getConfigValue("language");
+            if (config == null)
+            {
+                config = new Config();
+                config.WriteConfigFile();

```

Comment:
Because in previous versions, config was loaded after UI initialization.
So I just load 'language' in here.
If 'config.xml' is created/initialized here, the data cannot be configured after the program is opened.
I think we can reconstruct the methods of class config, such as add insertion, deletion, search, modification and so on.
After this we just initialize 'language' in here. At this point only one configuration data may exist in the configuration file.Other data is only written when needed.
Or we can add a setting that records that the user is using Sqlite or Mysql.

Reply:
@AyaseCore Fixed in e87876a6314671d5e67df3991c40b00613ad4f1b. Thanks for pointing this out - I had not noticed that the SQLite/MySQL selection had stopped popping up, and upon testing without a config found it just writes the default values. 😁 

The whole config creation/reading needs rewriting really. We can switch between connection mode with a dropdown like the language, and have another tab panel or a new window for configuring the different database connection types.